core/src/main/kotlin/xtdb/cache/MemoryCache.kt

```diff
@@ -34,7 +47,23 @@ class MemoryCache
 ) : AutoCloseable {
     private val pinningCache = PinningCache<PathSlice, Entry>(maxSizeBytes)
 
-    val stats get() = pinningCache.stats
+    val stats: MemoryCacheStats
+        get()  {
```

Comment:
I'm assuming you considered keeping running tallies of these as mutable state? don't know how expensive this might be if the cache's large

(might not be an issue, of course)

Reply:
I didn't want to to do that on the hot paths.

Reply:
What exactly do you mean by running tallies? Like for example a diff of files previously in the cache and now in the cache? This could be done on `getStats`.